lib/rules/media-feature-name-no-unknown/index.js

```diff
@@ -45,52 +53,127 @@ const rule = (primary, secondaryOptions) => {
 		}
 
 		root.walkAtRules(/^media$/i, (atRule) => {
-			mediaParser(atRule.params).walk(/^media-feature$/i, (mediaFeatureNode) => {
-				const parent = mediaFeatureNode.parent;
-				const mediaFeatureRangeContext = isRangeContextMediaFeature(parent.value);
-
-				let value;
-				let sourceIndex;
-
-				if (mediaFeatureRangeContext) {
-					const parsedRangeContext = rangeContextNodeParser(mediaFeatureNode);
-
-					value = parsedRangeContext.name.value;
-					sourceIndex = parsedRangeContext.name.sourceIndex;
-				} else {
-					value = mediaFeatureNode.value;
-					sourceIndex = mediaFeatureNode.sourceIndex;
-				}
-
-				if (!isStandardSyntaxMediaFeatureName(value) || isCustomMediaQuery(value)) {
+			/**
+			 * @param {string} featureName
+			 * @param {number} startIndex
+			 * @param {number} endIndex
+			 */
+			const validateFeatureName = (featureName, startIndex, endIndex) => {
```

Comment:
[suggest] Moving this `validateFeatureName` function out of the `root.walkAtRules()` callback may improve code readability like this:

```js
/**
 * @param {import('postcss').AtRule} atRule
 * @param {string} featureName
 * @param {number} startIndex
 * @param {number} endIndex
 */
const validateFeatureName = (atRule, featureName, startIndex, endIndex) => {
	// ...
};

root.walkAtRules(/^media$/i, (atRule) => {
	// ...

	validateFeatureName(atRule, node.getName(), startIndex, endIndex);

	// ...
});
```

What do you think?

Reply:
Yes, this is better!
Thank you.